src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts

```diff
@@ -461,8 +464,23 @@ export class ReadingsGraphComponent implements OnDestroy {
           r: 10,
           t: 25
         }
+      },
+      config: {
+        displayModeBar: false
       }
     };
+    this.generate3Dgraph();
+  }
+
+  public async generate3Dgraph() {
+    const intervalId = setInterval(() => {
```

Comment:
Could you please put a elaborating comment here, why 100 ms interval based retries are required?